src/Service/JsonSchemaValidator.php

```diff
@@ -0,0 +1,120 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Atoolo\Form\Service;
+
+use Atoolo\Form\Service\JsonSchemaValidator\Constraint;
+use Atoolo\Form\Service\JsonSchemaValidator\Extended\ValidatorExtended;
+use Atoolo\Form\Service\JsonSchemaValidator\FormatConstraint;
+use InvalidArgumentException;
+use JsonException;
+use LogicException;
+use Opis\JsonSchema\Errors\ErrorFormatter;
+use Opis\JsonSchema\Errors\ValidationError;
+use Opis\JsonSchema\Validator;
+use Symfony\Component\DependencyInjection\Attribute\Autowire;
+use Symfony\Component\DependencyInjection\Attribute\AutowireIterator;
+use Symfony\Component\Validator\Constraints\NotBlank;
+use Symfony\Component\Validator\ConstraintViolation;
+use Symfony\Component\Validator\ConstraintViolationList;
+use Symfony\Component\Validator\Exception\ValidationFailedException;
+
+class JsonSchemaValidator
+{
+    /**
+     * @param iterable<Constraint> $constraints
+     */
+    public function __construct(
+        #[Autowire('@' . ValidatorExtended::class)]
+        private readonly Validator $validator,
+        #[AutowireIterator('atoolo_form.jsonSchemaConstraint')]
+        iterable $constraints,
+        private readonly Platform $platform,
+    ) {
+        foreach ($constraints as $constraint) {
+            $this->registerConstraint($constraint);
+        }
+    }
+
+    private function registerConstraint(Constraint $constraint): void
+    {
+        if ($constraint instanceof FormatConstraint) {
+            $this->registerFormatConstraint($constraint);
+        } else {
+            throw new InvalidArgumentException('Unknown constraint type ' . get_class($constraint));
+        }
+    }
+
+    private function registerFormatConstraint(FormatConstraint $constraint): void
+    {
+        $formatResolver = $this->validator->parser()->getFormatResolver();
+        if ($formatResolver === null) {
+            throw new LogicException('No format resolver found');
+        }
+
+        $type = $constraint->getType();
+        $name = $constraint->getName();
+
+        $formatResolver->registerCallable(
+            $type,
+            $name,
+            function ($data, $schema) use ($constraint) {
+                return $constraint->check($data, $schema);
+            },
```

Comment:
this function does not need to bind `$this`, it should probably be `static`.

Reply:
Here I don't understand what you mean. Where is `$this` bind.